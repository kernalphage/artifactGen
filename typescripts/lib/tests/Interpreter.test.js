"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Scanner_1 = require("../core/Scanner");
const Parser_1 = require("../core/Parser");
const Interpreter_1 = require("../core/Interpreter");
test('basic json value', () => {
    let source = `
   [item]
   description = this is a @metalColor ring that is worth @metalValue;
   pi = 3.14;
   randomItem = 1:10;
   metalColor, metalValue = gold, 200 | silver, 100 | copper, 10 ;
  
   [Second]
   value = "special 324";
   value2 = "324"
  `;
    let scan = new Scanner_1.Scanner(source);
    let parse = new Parser_1.Parser(scan.tokens);
    let basic = new Interpreter_1.BasicInterpreter();
    basic.execute(parse.definitions);
    expect(basic.export()).toMatchSnapshot();
});
