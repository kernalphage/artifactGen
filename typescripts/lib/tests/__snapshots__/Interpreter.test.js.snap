// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic json value 1`] = `""`;
